tests/unit/lib/Any_unittest.cc

```diff
@@ -636,3 +636,10 @@ TEST(Any, reset_array2d_reference)
 	EXPECT_EQ(obj->ref_count(), 1);
 	SG_UNREF(obj);
 }
+
+TEST(Any, lazy_simple)
+{
+	auto v = 9;
+	auto any = make_any<int>([=]() { return v; });
+	EXPECT_EQ(any.as<int32_t>(), v);
```

Comment:
@karlnapf this thing works now, but it definitely needs more thorough testing

Reply:
shall we try to test a slightly more elaborate example of registering a member function that returns a constant or something?

Reply:
Yeah I can add one.